src/librustc/lint/builtin.rs

```diff
@@ -2036,6 +2036,35 @@ impl LintPass for HardwiredLints {
     }
 }
 
+declare_lint! {
+    UNEXPORTED_NO_MANGLE,
```

Comment:
Yeah out of this I think that `private-no-mangle-fns` is the best option. I agree with keeping `no-mangle` in the name and keeping `priv` out.